pkg/action/upgrade.go

```diff
@@ -130,7 +132,7 @@ func (u *Upgrade) prepareUpgrade(name string, chart *chart.Chart, vals map[strin
 
 	// finds the deployed release with the given name
 	currentRelease, err := u.cfg.Releases.Deployed(name)
-	if err != nil {
+	if err != nil && err != driver.ErrReleaseNotFound {
```

Comment:
are you sure you want to ignore the not found errors here?

Reply:
Yea, the tests need to be fixed for this.

Reply:
Fixed per #7372's handling of mutable data.